tests/platform_tests/test_process_reboot_cause.py

```diff
@@ -0,0 +1,190 @@
+                main_pid = int(line.split("=")[-1])
+            elif "ExecMainStatus" in line:
+                status = int(line.split("=")[-1])
+            elif "Result" in line:
+                result = line.split("=")[-1]
+
+        return main_pid, status, result
+
+    def restart_service(self):
+        logging.info("Restarting process-reboot-cause.service")
+        _ = self.duthost.command(SYSTEMCTL_RESTART_CMD)
+
+    def create_json_file(self, filetype=""):
+
+        gen_time = self.duthost.command(REBOOT_CAUSE_GEN_TIME_CMD)["stdout_lines"][0]  # noqa: E501
+        reboot_cause_file = os.path.join(REBOOT_CAUSE_HISTORY_DIR, REBOOT_CAUSE_FILE_FORMAT.format(gen_time))  # noqa: E501
+
+        logging.info("Writing '{}' to {} on DUT".format(filetype, reboot_cause_file))  # noqa: E501
+        _ = self.duthost.command("bash -c 'echo {}>{}'".format(filetype, reboot_cause_file))  # noqa: E501
+
+        return reboot_cause_file
+
+    def remove_json_file(self, reboot_cause_file):
+        logging.info("Clean up the newly created JSON file: {}".format(reboot_cause_file))  # noqa: E501
+        _ = self.duthost.command('rm -f {}'.format(reboot_cause_file))
+
+    def bad_json_file_assertions(self, status, result, journalctl_output):
+
+        # Check for exeception if image version < 2023110.20 , 20230531.33 or 20220531.53  # noqa: E501
+        if self.image_major_ver < 20220531 or \
+          (self.image_major_ver == 20231110 and self.image_minor_ver < 20) or \
+          (self.image_major_ver == 20230531 and self.image_minor_ver < 33) or \
+          (self.image_major_ver == 20220531 and self.image_minor_ver < 53):
+
+            logging.info("OS Version {} does not have the patched process-reboot-cause file with try-except block".format(self.image_ver))  # noqa: E501
+            is_json_decode_error = any(JSON_DECODE_ERROR in line for line in journalctl_output)  # noqa: E501
+
+            pytest_assert(result == "exit-code")
+            pytest_assert(status == 1)
+            pytest_assert(is_json_decode_error)
+
+        else:
+            logging.info("OS Version {} has the patched process-reboot-cause file with try-except block".format(self.image_ver))  # noqa: E501
+            exception_handled_log = any(EXCEPTION_HANDLED_SYSLOG in line for line in journalctl_output)  # noqa: E501
+            logging.info("exception_handled_log: {}".format(journalctl_output))
+            pytest_assert(result == "success")
+            pytest_assert(status == 0)
+            pytest_assert(exception_handled_log)
+
+    def good_json_file_assertions(self, status, result, cause, journalctl_output, statedb_output):  # noqa: E501
+
+        excpected_reboot_cause = any(cause in line for line in statedb_output)  # noqa: E501
+
+        pytest_assert(result == "success")
+        pytest_assert(status == 0)
+        pytest_assert(excpected_reboot_cause)
+
+    def process_reboot_cause_file_runner(self, filetype=""):
```

Comment:
@assrinivasan please add brief comment what is this checking?

Reply:
Done in latest